hax/hax/motr/__init__.py

```diff
@@ -126,14 +126,12 @@ def _entrypoint_request_cb(self, reply_context: Any, req_id: Any,
                 # anyway detect the failure and report the same so we exclude
                 # reporting the same during their first entrypoint request.
                 # But we need to do it for motr server processes.
-                q: Queue = Queue(1)
+                # q: Queue = Queue(1)
                 LOG.debug('first entrypoint request, broadcasting FAILED')
-                self.queue.put(
-                    BroadcastHAStates(
-                        states=[HAState(fid=process_fid,
-                                        status=ServiceHealth.FAILED)],
-                        reply_to=q))
-                ids: List[MessageId] = q.get()
+                states = [HAState(fid=process_fid,
```

Comment:
Do you have an idea why you experience a delay? Can't it be a symptom that the Queue capacity is too low and we need to enlarge it?

Seriously, initially the queue size was chosen quite small - 8 objects or so. I can imagine that under a high load (when there are many Motr processes to communicate to and thus many events are happening, the size can be a bottleneck.

The idea is that when the Queue is full and a thread is trying to add a message there, the thread is blocked until some free slot in the queue becomes available.

Reply:
Another idea: if we do depend on the speed of reaction from hax, we can add more "consumer" threads. But before doing that we need to make sure that this doesn't bring other side effects.

For instance, currently all Motr-related jobs are processed one-by-one in the order they were required. If we start processing them in parallel, can this bring issues to Motr world? If there is no issue for sure, I think we should enlarge the number of threads.

@andriytk, @max-seagate, @chumakd Thoughts?